components/component.go

```diff
@@ -41,6 +41,10 @@ func (c *Component) GetManagementState() operatorv1.ManagementState {
 	return c.ManagementState
 }
 
+func (c *Component) SetManagementState(state operatorv1.ManagementState) {
```

Comment:
how will this be used except for the test part.
say:
`k.SetManagementState(operatorv1.Managed) `done in reconcile of kserve, then it will not respect the value set in the DSC CR for kserve?

Reply:
Valid point. In the testing code the call is simulating user input, but in production the object is user input.

What would you recommend for the testing code?  I can use reflection to set the value.

Reply:
I'm convinced that it's wrong. About how I planned to use it, there is a 'request for comments' try to rewrite the tests https://github.com/opendatahub-io/opendatahub-operator/pull/880/files#diff-8d48bac6a439e62ea3b0740b26d2d6b9c93e1aeebc67aa9cc562abac695b13d5R611 

Whatever the faith of the proposal is, we discussed with QA guys about testcases for different combinations of enable/disable components instead of static configuration. So some sort of that functionality would be useful.

But no problem, I'll implement it inside.

Reply:
> I'm convinced that it's wrong

What is "that" you are referring to?

>  different combinations of enable/disable components instead of static configuration. So some sort of that functionality would be useful.

I am not sure I am on the same page here. What is the benefit of having setter `k.SetManagementState(Removed)` when you already can call it `k.ManagementState = Removed`?

As for #880 - if you want to open a Pandora box go ahead and formally add me as a reviewer there :)

Reply:
> > I'm convinced that it's wrong
> 
> What is "that" you are referring to?

To make such a public method for the structure.

> 
> As for #880 - if you want to open a Pandora box go ahead and formally add me as a reviewer there :)

You are welcome for sure :)